app/src/main/java/org/mozilla/fenix/library/history/HistorySearchDialogFragment.kt

```diff
@@ -0,0 +1,261 @@
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.toolbar.ToolbarPosition
+import org.mozilla.fenix.databinding.FragmentHistorySearchDialogBinding
+import org.mozilla.fenix.databinding.SearchSuggestionsHintBinding
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.library.history.awesomebar.AwesomeBarView
+import org.mozilla.fenix.library.history.toolbar.ToolbarView
+import org.mozilla.fenix.settings.SupportUtils
+
+@Suppress("TooManyFunctions")
+class HistorySearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
+    private var _binding: FragmentHistorySearchDialogBinding? = null
+    private val binding get() = _binding!!
+
+    private lateinit var interactor: HistorySearchDialogInteractor
+    private lateinit var store: HistorySearchFragmentStore
+    private lateinit var toolbarView: ToolbarView
+    private lateinit var awesomeBarView: AwesomeBarView
+
+    private var dialogHandledAction = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(STYLE_NO_TITLE, R.style.SearchDialogStyle)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return object : Dialog(requireContext(), this.theme) {
+            override fun onBackPressed() {
+                this@HistorySearchDialogFragment.onBackPressed()
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentHistorySearchDialogBinding.inflate(inflater, container, false)
+        val activity = requireActivity() as HomeActivity
+
+        store = HistorySearchFragmentStore(
+            createInitialHistorySearchFragmentState()
+        )
+
+        interactor = HistorySearchDialogInteractor(
+            HistorySearchDialogController(
+                activity = activity,
+                fragmentStore = store,
+                clearToolbarFocus = {
+                    dialogHandledAction = true
+                    toolbarView.view.hideKeyboard()
+                    toolbarView.view.clearFocus()
+                },
+            )
+        )
+
+        toolbarView = ToolbarView(
+            context = requireContext(),
+            interactor = interactor,
+            isPrivate = false,
+            view = binding.toolbar,
+        )
+
+        val awesomeBar = binding.awesomeBar
+
+        awesomeBarView = AwesomeBarView(
+            activity,
+            interactor,
+            awesomeBar,
+        )
+
+        awesomeBarView.view.setOnEditSuggestionListener(toolbarView.view::setSearchTerms)
+        binding.searchWrapper.background = ColorDrawable(Color.TRANSPARENT)
```

Comment:
why set this at runtime instead of in the XML?

Reply:
Actually this is something I forgot to remove. I'll remove it.  Thanks,